### ISSUE TITLE
a11y: Talkback doesn't navigate to offscreen options when magnification is on

### ISSUE THREAD

**Issue by jkurtw** [flutter/flutter#49322]
Internal: b/148171397

Using a bottom sheet with two buttons, if you turn on magnification, then the buttons cannot be selected via TalkBack. 

This should be only on Android. I cannot reproduce this on iOS.

Here is the sample code:
```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Welcome to Flutter',
      theme: ThemeData(primaryColor: Colors.white),
      home: HomePage(),
    );
  }
}

class HomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: FlatButton(
        child: Center(child: Text("Show")),
        onPressed: () {
          showModalBottomSheet(
            context: context,
            backgroundColor: Colors.transparent,
            builder: (_) => _buildRoundedBottomSheet(
              context,
              _buildConfirmationPage(context),
            ),
          );
        },
      ),
    );
  }

  Widget _buildConfirmationPage(BuildContext context) {
    return Container(
      padding: EdgeInsets.fromLTRB(24, 16, 24, 0),
      child: SingleChildScrollView(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            Flexible(child: Text("Delete this information?")),
            SizedBox(height: 8),
            Flexible(child: Text("Are you sure?")),
            SizedBox(height: 8),
            Wrap(
              direction: Axis.horizontal,
              alignment: WrapAlignment.end,
              runSpacing: 8,
              children: [
                FlatButton(
                  padding: EdgeInsets.all(8.0),
                  onPressed: () {},
                  shape: const RoundedRectangleBorder(),
                  child: Text("Cancel"),
                ),
                FlatButton(
                  padding: EdgeInsets.all(8.0),
                  onPressed: () {},
                  shape: const RoundedRectangleBorder(),
                  child: Text("Continue"),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildRoundedBottomSheet(BuildContext context, Widget child) {
    return Container(
      decoration: BoxDecoration(
        color: Theme.of(context).scaffoldBackgroundColor,
        borderRadius: BorderRadius.vertical(top: Radius.circular(16)),
      ),
      padding: const EdgeInsets.symmetric(vertical: 16),
      child: Stack(
        children: [child],
      ),
    );
  }
}
```

**Comment by darrenaustin** [comment]
I am able to reproduce the problem with given code (thanks for that). 

It does appear to work with magnification on if you pan the screen down such that the bottom sheet is visible. Then it appears to allow a11y focus to move to it. So that may be a workaround.

I'm not sure what is causing the issue here. Perhaps the magnification is throwing off the active regions we are reporting for each control, or they are being interpreted differently.

@cbracken can you take a look at this and see if the engine's Android bridge is setting the right bounds for the controls in the bottom sheet when they aren't on the screen?  If this does appear to be a framework bug, please pass it back to me. Thx.

**Comment by cbracken** [comment]
I can partially repro this. 

Things that do not work but should:
* Accessibility focus ring doesn't show onscreen. Verified this _does_ work in apps like Calculator.

Things that do work for me:
* The current accessibility focus is correct.
* Talkback reads out the currently focused item.
* I can use swipe navigation to move the accessibility focus (though the ring doesn't show)
* I can use double-tap to invoke actions such as tap.

FWIW I enabled magnification via the triple-tap mechanism.

This seems likely to be an issue in the Android a11y bridge where we're likely failing to apply the correct logical to screen coordinate matrix transforms. Will update when I've got a little more.

**Comment by pcsosinski** [comment]
any update @cbracken?

**Comment by xster** [comment]
@cbracken are you currently working on this?

**Comment by jonahwilliams** [comment]
The magnification is performed on a pixel level at the screen and has no impact on the semantics/accessibility we report, it is also a separate service from TalkBack. I'm going to do some googling today to see if there are any mentions of magnification in the talkback source

**Comment by jonahwilliams** [comment]
Err, it might not be a separate service - stay tuned

**Comment by jonahwilliams** [comment]
here is the magnification logic that navigates to offscreen:

https://github.com/google/talkback/blob/9b1d5132b074174d02886faccf731e814d69363c/talkback/src/main/java/eventprocessor/ProcessorMagnification.java

**Comment by jonahwilliams** [comment]
Err, thats not quite right.

Perhaps whatever event this is looking for doesn't have a node attached: https://github.com/google/talkback/blob/9b1d5132b074174d02886faccf731e814d69363c/talkback/src/main/java/eventprocessor/ProcessorMagnification.java#L63

**Comment by jonahwilliams** [comment]
yeah, we always set this to the root view: https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L1585

**Comment by jonahwilliams** [comment]
wait nope, nevermind - it is the right value

**Comment by xster** [comment]
@jonahwilliams tentatively assigning to you since it looks like you're making some progress.

**Comment by jonahwilliams** [comment]
I'm not planning on making any more progress, I've commented on the buganizer issue with my findings

**Comment by xster** [comment]
FWIW, I was able to get it to navigate anywhere on screen zoomed in in a normal Android share screen. But it is indeed very jumpy.

**Comment by chunhtai** [comment]
I can't seems to reproduce it using OnePlus, may i know which device were you testing on?

**Comment by pcsosinski** [comment]
any update on a repro here?

**Comment by chinmaygarde** [comment]
We are trying to find who owns this issue internally. @pcsosinski: Assigning to you per Rays guidance.
